restapi.go

```diff
@@ -41,13 +41,22 @@ var (
 	ErrUnauthorized            = errors.New("HTTP request was unauthorized. This could be because the provided token was not a bot token. Please add \"Bot \" to the start of your token. https://discordapp.com/developers/docs/reference#authentication-example-bot-token-authorization-header")
 )
 
+type requestHeader struct {
+	key, val string
+}
+
 // Request is the same as RequestWithBucketID but the bucket id is the same as the urlStr
 func (s *Session) Request(method, urlStr string, data interface{}) (response []byte, err error) {
 	return s.RequestWithBucketID(method, urlStr, data, strings.SplitN(urlStr, "?", 2)[0])
 }
 
-// RequestWithBucketID makes a (GET/POST/...) Requests to Discord REST API with JSON data.
+// RequestWithBucketID is the same as requestWithBucketID but does not support optional headers.
 func (s *Session) RequestWithBucketID(method, urlStr string, data interface{}, bucketID string) (response []byte, err error) {
+	return s.requestWithBucketID(method, urlStr, data, bucketID, nil)
+}
+
+// requestWithBucketID makes a (GET/POST/...) Requests to Discord REST API with JSON data.
```

Comment:
Perhaps add something about the optional headers in this comment (going along with the above review comment)